opt_adapt/opt.py

```diff
@@ -52,11 +53,33 @@ def __init__(self, options={}):
                 raise ValueError(f"Option {key} not recognised")
             self.__setattr__(key, value)
 
+# Only works for problem with control paramter in r-space
+def line_search(forward_run, mesh, u, P, J, dJ, Rspace, alpha=1e-4, max_search_iter=100):
+    
+    lr = 1
```

Comment:
I think it would be a good idea to pass `params` as an input and then read `params.lr` from there?

Reply:
That way the user can specify a smaller or larger initial guess, if that is useful.